apps/api/src/report/report.service.ts

```diff
@@ -221,37 +221,149 @@ export class ReportService {
+            newProcessApplicants: licenceResults.newProvisionalLicence[0]?.count || '0',
+          };
+        case 'Applied to NNAS':
+          return {
+            status: step,
+            oldProcessApplicants: this.findAndSumStatusCounts(oldProcess, [
+              statuses[STATUS.APPLIED_TO_NNAS],
+              statuses[STATUS.SUBMITTED_DOCUMENTS],
+              statuses[STATUS.RECEIVED_NNAS_REPORT],
+            ]),
+            newProcessApplicants: this.findAndSumStatusCounts(newProcess, [
+              statuses[STATUS.APPLIED_TO_NNAS],
+              statuses[STATUS.SUBMITTED_DOCUMENTS],
+              statuses[STATUS.RECEIVED_NNAS_REPORT],
+            ]),
+          };
 
-    const newProcess = await entityManager.query(
-      this.reportUtilService.licensingStageApplicantsQuery(statuses, from, to, true),
-    );
+        case 'Completed Additional Education':
+          return {
+            status: step,
+            oldProcessApplicants: this.findAndSumStatusCounts(oldProcess, [
+              statuses[STATUS.REFERRED_TO_ADDITIONAL_EDUCTION],
+              statuses[STATUS.COMPLETED_ADDITIONAL_EDUCATION],
+            ]),
+            newProcessApplicants: this.findAndSumStatusCounts(newProcess, [
+              statuses[STATUS.REFERRED_TO_ADDITIONAL_EDUCTION],
+              statuses[STATUS.COMPLETED_ADDITIONAL_EDUCATION],
+            ]),
+          };
 
-    // combine old data with new data
-    const data = oldProcess.map((o: { status: string; applicants: string }) => {
-      const newProcessApplicants = newProcess.find(
-        (n: { status: string }) => n.status === o.status,
+        case 'Referred to NCAS':
+          return {
+            status: step,
+            oldProcessApplicants: this.findAndSumStatusCounts(oldProcess, [
+              statuses[STATUS.REFERRED_TO_NCAS],
+              statuses[STATUS.COMPLETED_CBA],
+              statuses[STATUS.COMPLETED_SLA],
+            ]),
+            newProcessApplicants: this.findAndSumStatusCounts(newProcess, [
+              statuses[STATUS.REFERRED_TO_NCAS],
+              statuses[STATUS.COMPLETED_CBA],
+              statuses[STATUS.COMPLETED_SLA],
+            ]),
+          };
+        case 'Withdrew from IEN program':
+          return {
+            status: step,
+            oldProcessApplicants: withdrawnOld,
+            newProcessApplicants: withdrawnNew,
+          };
+        default:
+          return {
+            status: step,
+            oldProcessApplicants:
+              oldProcess.find(value => value.status_id === statuses[step])?.count || '0',
+            newProcessApplicants:
+              newProcess.find(value => value.status_id === statuses[step])?.count || '0',
+          };
+      }
+    });
+    return report;
+  }
+  findAndSumStatusCounts(list: { status_id: string; count: string }[], acceptedIds: string[]) {
+    let count = 0;
+    acceptedIds.forEach(id => {
+      count += parseInt(
```

Comment:
`+` instead of parseInt? only because we've been using it throughout the project already